lib/util.js

```diff
@@ -613,13 +613,18 @@ function formatValue(ctx, value, recurseTimes) {
     }
   }
 
-  ctx.seen.push(value);
+  // TODO(addaleax): Make `seen` a Set to avoid linear-time lookup.
+  if (ctx.seen.includes(value)) {
+    return ctx.stylize('[Circular]', 'special');
+  } else {
```

Comment:
any reason to add the `else` here? The diff would be smaller without it. Just a nit though